api/models.py

```diff
@@ -1,38 +1,66 @@
-from django.db import models
 from django.contrib.auth.models import User
+from django.db import models
+from django.db.models import F, Q
+
 
-# Create your models here.
 class Video(models.Model):
     title = models.CharField(max_length=100)
     created_at = models.DateTimeField(auto_now_add=True)
     viewed_at = models.DateTimeField(null=True)
 
-#All possible relationships
-#(true,true), (true, null), (false, true/null/false), (null,null)
-#Note that we don't need to store (null,null)
+
 class MatchStatus(models.Model):
-    primary_user = models.ForeignKey(User, on_delete=models.CASCADE, related_name="+")
-    secondary_user = models.ForeignKey(User, null=True, on_delete=models.CASCADE, related_name="+")
-    primary_response = models.BooleanField()
-    secondary_response = models.BooleanField(null=True)
+    """
+    Represents the match status for two users.
+
+    A True response means that the user accepted the other user as a penpal.
+    A False response means that the user declined the other user as a penpal.
+    A null response means that the user is undecided.
+    """
+
+    user_lo = models.ForeignKey(
+        User, on_delete=models.CASCADE, related_name="matchstatus_lo"
+    )
+    user_hi = models.ForeignKey(
+        User, on_delete=models.CASCADE, related_name="matchstatus_hi"
+    )
+    user_lo_response = models.BooleanField(null=True)
+    user_hi_response = models.BooleanField(null=True)
+
+    class Meta:
+        constraints = [
+            # This constraint avoids duplicate/illegal records:
+            #   user1  user2   . . .
+            #   -----  -----
+            #   0034   0035     }
+            #   0035   0034     } Only one of these two records needs to exist
+            #   . . .
+            #   0022   0022     Users can't match with themselves
+            models.CheckConstraint(
+                check=Q(user_lo_id__lt=F("user_hi_id")), name="user_lo_lt_user_hi"
```

Comment:
Why was F needed here?

Reply:
You need to use F to reference the value of the column from the table

Reply:
Why does `user_lo_id < user_hi_id` prevent self-matching? To me this introduces a weird side-affect

Reply:
Because if `user_lo_id == user_hi_id`, this would violate the constraint and not be allowed into the database

Reply:
It doesn't directly prevent self-matching. It's more of a database integrity thing.

Reply:
So why isn't it `check=~Q(user_lo_id=F("user_hi_id")), name="user_lo_neq_user_hi"

Reply:
Because we also don't want entries where the two user ids are swapped because they mean the same thing.

Reply:
What do you mean by swapped?

Reply:
Ah I understand now, you're working to avoid duplicates as well. Now this makes sense